apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -87,20 +82,16 @@ def delete(self) -> None:
         """
 
         # TODO: should this raise an error if the object does not exist?
+        self.close()
         try:
+            self._was_deleted = True
             tiledb.remove(self._uri)
         except tiledb.TileDBError:
             pass
         return
 
     def __repr__(self) -> str:
-        """
-        Default repr
-        """
-        if self.exists():
-            return f'{self.soma_type}(uri="{self._uri}")'
-        else:
-            return f"{self.soma_type}(not created)"
+        return f'{self.soma_type}(uri="{self._uri}")'
```

Comment:
I can't express how much joy I feel with the removal of the `(not created)` string!